src/react.ts

```diff
@@ -2,18 +2,17 @@ import { useDebugValue } from 'react'
 import { useSyncExternalStoreWithSelector } from 'use-sync-external-store/shim/with-selector'
 import createStore, {
   EqualityChecker,
-  GetState,
-  SetState,
+  Mutate,
   State,
   StateCreator,
   StateSelector,
   StoreApi,
+  StoreMutatorIdentifier,
 } from './vanilla'
 
-export function useStore<T extends State>(api: StoreApi<T>): T
-export function useStore<T extends State, U>(
-  api: StoreApi<T>,
-  selector: StateSelector<T, U>,
+export function useStore<S extends StoreApi<State>, U = ExtractState<S>>(
```

Comment:
I thinks it's rare that people specify `useStore<MyState>(...)`, but as this is a breaking change, it's worth noting in release note. Can you create a summary of public type changes in the description of this PR please?

Reply:
Yes it indeed is a breaking change. Sure, I'll create a summary soon.